File: apps/web/test/fixtures/catalog.ts
import type { PageSpeciesResponse, PagePlantVarietyResponse, PlantCategoryResponse, PlantVarietyResponse, SpeciesResponse } from "@/lib/api/catalog";

/** Shared fixtures for 7F Plant Catalog tests -- mirrors test/fixtures/organization.ts's pattern. */

export function makePlantCategory(overrides: Partial<PlantCategoryResponse> = {}): PlantCategoryResponse {
  return {
    id: "99999999-9999-9999-9999-999999999901",
    code: "houseplant",
    name: "Houseplant",
    description: "Indoor foliage and flowering plants.",
    ...overrides,
  };
}

export function makeSpecies(overrides: Partial<SpeciesResponse> = {}): SpeciesResponse {
  return {
    id: "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaa01",
    nursery_id: "22222222-2222-2222-2222-222222222222",
    category_id: "99999999-9999-9999-9999-999999999901",
    common_name: "Fiddle Leaf Fig",
    botanical_name: "Ficus lyrata",
    light_requirement: "bright_indirect",
    water_baseline_ml_per_week: 500,
    soil_type: "well_draining",
    temperature_min_celsius: 16,
    temperature_max_celsius: 27,
    growth_curve_baseline: [{ days_since_planting: 0, expected_height_cm: 30 }],
    disease_susceptibility: ["root rot"],
    created_at: "2026-01-01T00:00:00Z",
    updated_at: "2026-01-01T00:00:00Z",
    ...overrides,
  };
}

export function makeSpeciesPage(items: SpeciesResponse[] = [makeSpecies()]): PageSpeciesResponse {
  return {
    items,
    meta: { page: 1, page_size: 20, total_items: items.length, total_pages: 1 },
  };
}

export function makePlantVariety(overrides: Partial<PlantVarietyResponse> = {}): PlantVarietyResponse {
  return {
    id: "bbbbbbbb-bbbb-bbbb-bbbb-bbbbbbbbbb01",
    nursery_id: "22222222-2222-2222-2222-222222222222",
    species_id: "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaa01",
    name: "Bambino",
    description: "A dwarf cultivar.",
    created_at: "2026-01-01T00:00:00Z",
    updated_at: "2026-01-01T00:00:00Z",
    ...overrides,
  };
}

export function makePlantVarietyPage(items: PlantVarietyResponse[] = [makePlantVariety()]): PagePlantVarietyResponse {
  return {
    items,
    meta: { page: 1, page_size: 100, total_items: items.length, total_pages: 1 },
  };
}

File: apps/api/app/schemas/catalog.py
"""Pydantic request/response DTOs for Module 5's Species Catalog (FR-4)."""
from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class PlantCategoryResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    code: str
    name: str
    description: str | None


class GrowthCurvePoint(BaseModel):
    days_since_planting: int = Field(..., ge=0)
    expected_height_cm: float = Field(..., ge=0)


class CreateSpeciesRequest(BaseModel):
    category_id: uuid.UUID
    common_name: str = Field(..., min_length=1, max_length=255)
    botanical_name: str = Field(..., min_length=1, max_length=255)
    light_requirement: str | None = Field(None, max_length=50)
    water_baseline_ml_per_week: int | None = Field(None, ge=0)
    soil_type: str | None = Field(None, max_length=100)
    temperature_min_celsius: float | None = None
    temperature_max_celsius: float | None = None
    growth_curve_baseline: list[GrowthCurvePoint] | None = None
    disease_susceptibility: list[str] | None = None


class UpdateSpeciesRequest(BaseModel):
    category_id: uuid.UUID | None = None
    common_name: str | None = Field(None, min_length=1, max_length=255)
    botanical_name: str | None = Field(None, min_length=1, max_length=255)
    light_requirement: str | None = Field(None, max_length=50)
    water_baseline_ml_per_week: int | None = Field(None, ge=0)
    soil_type: str | None = Field(None, max_length=100)
    temperature_min_celsius: float | None = None
    temperature_max_celsius: float | None = None
    growth_curve_baseline: list[GrowthCurvePoint] | None = None
    disease_susceptibility: list[str] | None = None


class SpeciesResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    nursery_id: uuid.UUID
    category_id: uuid.UUID
    common_name: str
    botanical_name: str
    light_requirement: str | None
    water_baseline_ml_per_week: int | None
    soil_type: str | None
    temperature_min_celsius: float | None
    temperature_max_celsius: float | None
    growth_curve_baseline: list | None
    disease_susceptibility: list | None
    created_at: datetime
    updated_at: datetime


class CreatePlantVarietyRequest(BaseModel):
    species_id: uuid.UUID
    name: str = Field(..., min_length=1, max_length=255)
    description: str | None = Field(None, max_length=500)


class UpdatePlantVarietyRequest(BaseModel):
    name: str | None = Field(None, min_length=1, max_length=255)
    description: str | None = Field(None, max_length=500)


class PlantVarietyResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    nursery_id: uuid.UUID
    species_id: uuid.UUID
    name: str
    description: str | None
    created_at: datetime
    updated_at: datetime
